rbx_binary/src/serializer.rs

```diff
@@ -977,6 +1028,23 @@ impl<'a, W: Write> BinarySerializer<'a, W> {
 
                         chunk.write_interleaved_i64_array(buf.into_iter())?;
                     }
+                    Type::SharedString => {
+                        let mut entries = Vec::with_capacity(values.len());
+
+                        for (i, rbx_value) in values {
+                            if let Variant::SharedString(value) = rbx_value.as_ref() {
+                                let id = self
+                                    .shared_string_ids
+                                    .get(value)
+                                    .ok_or_else(|| InnerError::TooManySharedStrings {})?;
```

Comment:
Same comment as other spot, we shouldn't be worried about overflowing a u32 here.